### ISSUE TITLE
Enable searching for Flutter plugins specifying supported native platforms (e.g. Android, iOS)

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/pub-dev#187]
When searching among Flutter packages, support narrowing the search to only packages that work on a particular platform (android, ios, web, desktop)

**Comment by mit-mit** [comment]
Related bug for package detail page: https://github.com/dart-lang/pub-dartlang-dart/issues/144

**Comment by mit-mit** [comment]
Blocked on https://github.com/flutter/flutter/issues/10288

**Comment by kevmoo** [comment]
@mit-mit Moving this to M4. We can re-triage once there is a plan on the flutter side to provide the needed info

**Comment by isoos** [comment]
@mit-mit: what is the latest on detecting whether a package works on ios, android or both?

**Comment by kevmoo** [comment]
I marked this as M4 – since we have no plan in the short term to address this

**Comment by mit-mit** [comment]
Correct, we don't have platform support for this yet.

**Comment by MarkOSullivan94** [comment]
Any estimation on when this will arrive? Could imagine there'll be an increase in packages published the more popular Flutter becomes

**Comment by jaumard** [comment]
Also would be nice to be able to filter if it's Java/objC vs kotlin/swift as kotlin/swift is more painful to maintain and personally I try to avoid them

**Comment by jonasfj** [comment]
@jaumard, analyzing language usages would be interesting I filed #2154 for this. But it's definitely a separate issue, though it might mitigate this issue.

**Comment by MarkOSullivan94** [comment]
Would be perfect if there was a way to request help from people with iOS side for packages which only have Android side implemented and vice versa

**Comment by fmatosqg** [comment]
@jaumard what are the issues with kotlin?

**Comment by jaumard** [comment]
@fmatosqg when you have multiple flutter plugin using kotlin, if one have an old version of kotlin your all project doesn't compile anymore... Or the opposite one plugin is really up to date but not you, still you can't compile unless you migrate all your plugin to the same kotlin version (or acceptable range of versions) but maybe one plugin is not yours and you're blocked... 

It's exactly same story as swift. That's why I ban those languages for flutter plugins. And I've seen many plugins migrate to Java/objC from swift/kotlin because of that, because maintaining an open source plugin using those languages force you to update then just to have languages version up to date, even if there no new feature or bug fixing.

**Comment by fmatosqg** [comment]
@jaumard having worked with kotlin for a while in native android this is an interesting surprise, specially considering how strong backwards compatibility is. I'd be very interesting in seeing that failing for myself, can you post a link to such a project?

**Comment by jaumard** [comment]
@fmatosqg I've worked a lot with kotlin myself and had no issues for android projects, but with flutter it's not the same story because plugins are loaded as "module" on our project, meaning they have they're own gradle/kotlin supported version and that's the problem as they can use a very old version and you'll not be able to compile anymore. 

So even if you are upgrading your project to the latest version, it doesn't mean that all the plugin you use will do too. I don't have a link to provide as it's my company project and it doesn't fail anymore as we migrate to only java/objC plugins to avoid those problems. but you can check `geolocator` or `flutter_crashlytics` who are plugin that has made the migration back to those languages

**Comment by MarkOSullivan94** [comment]
I feel like there should be some warning making everyone aware of what you're talking about @jaumard. Would be very useful whenever creating new projects and just informative on the Flutter docs.

You should open a PR on the Flutter website and add that key piece of information.

**Comment by jaumard** [comment]
I'm agree with, but I'm not sure it will be accepted @MarkOSullivan94 as saying "Swift/Kotlin is considered bad practice for Flutter plugin and should be avoided" might not be a good selling point ^^ 
I don't want to waste time making a PR for nothing :) but making a medium for good practice for Flutter plugins might be good yes (but maybe there already some ^^)

**Comment by jonasfj** [comment]
@jaumard, @MarkOSullivan94, whether to discourage swift/kotlin is a flutter concern, maybe @amirh have more insights on this. In any event it seems like something worth debating in a Flutter issue :)

**Comment by yjbanov** [comment]
Please also add the Web to the list of platforms to search by.

**Comment by yjbanov** [comment]
See, e.g. https://github.com/flutter/flutter/issues/40360
